Add support for user defined metadata

cloudnautique opened the pull request:
This PR adds support for user supplied metadata, that will be made
available to containers via the rancher-metadata api.

Minor changes were made to the test and bootstrap script to make the
/scratch code path work.

ibuildthecloud commented:
Can you change https://github.com/cloudnautique/rancher-compose/blob/c2dcfd5059f49e49bdee2549a1031c61f6508d3e/tests/integration/cattletest/core/test_compose.py#L1383 to `name='cert3'`.  The test was always wrong but the new cattle version checks for unique cert name now.

ibuildthecloud commented:
@cloudnautique And rebase one more time :)  Master was broken from the last hasty PR merge.

cloudnautique commented:
Updated the failing cert test. I added a wait_for call to make sure the cert is removed removed so that the name can be reused. It might be better to add a purged state to certificates? There isn't one according to our latest schema.

ibuildthecloud commented on tests/integration/cattletest/core/test_compose.py:
```diff
@@ -1369,23 +1380,26 @@ def test_cert_removed(new_context, compose_bin, request):
 
-    client.delete(cert2)
-    cert2 = client.wait_success(cert2)
+    cert2 = client.wait_success(cert2.remove())
```
Can you do `client.wait_success(cert2.purge())` does that not work.  Otherwise this test takes awhile just waiting for the cert to be removed.

cloudnautique replied:
There is no purge on the cert. So you have to wait the minute before its removed so you can use the name again.
